mnemosyne/core/beam.py

```
@@ -445,20 +466,18 @@ def _get_connection(db_path: Path = None) -> sqlite3.Connection:
-        or getattr(_thread_local, 'db_path', None) != str(path)
-    )
-    if not needs_reconnect:
+    key, path = _canonical_db_path(db_path)
+    connections = _connections()
+    conn = connections.get(key)
+    if conn is not None:
         # Verify the cached connection is still alive
         try:
-            _thread_local.conn.execute("SELECT 1")
+            conn.execute("SELECT 1")
         except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)